src/transforms/babel.js

```diff
@@ -27,25 +27,34 @@ const JSX_PRAGMA = {
   hyperapp: 'h'
 };
 
-async function babelTransform(asset) {
-  let config = await getConfig(asset);
+async function babelTransform(asset, version) {
+  let config = await getConfig(asset, version);
   if (!config) {
     return;
   }
 
   await asset.parseIfNeeded();
 
+  // Pre-Transform Babel 6
+  if (config.babelVersion && config.babelVersion === 6) {
+    let babel6 = await localRequire('babel-core', asset.name);
+    let res = babel6.transformFromAst(asset.ast, asset.contents, config);
+    if (res.ast) {
+      asset.ast = res.ast;
+      asset.isAstDirty = true;
+    }
+
+    asset.ast = babelASTConverter(asset.ast, 6);
+
+    return babelTransform(asset, 7);
```

Comment:
Won't this result in transforming every file twice?

Reply:
Yes, but it will only run babel 6 transforms on the first run, convert to babel 7 and run the internal config. Thought this was the expected behaviour.
I might have made a mistake though, not entirely sure this is the proper way to do this.

Reply:
oh I see. yeah that's how it should work.